packages/react-day-picker/src/contexts/Modifiers/ModifiersContext.tsx

```diff
@@ -16,74 +17,25 @@ export type ModifiersProviderProps = {
   children: React.ReactNode;
 };
 
-function toMatcherArray(modifierFromProp: Matcher | Matcher[] | undefined) {
-  if (Array.isArray(modifierFromProp)) {
-    return modifierFromProp;
-  } else if (modifierFromProp !== undefined) {
-    return [modifierFromProp];
-  } else {
-    return [];
-  }
-}
-
-/**
- * Parse the modifiers from the props.
- *
- * Internally we want modifiers as an array of matchers – as opposite of the
- * props which can accept also a matcher.
- */
+/** Provide the value for the [[ModifiersContext]]. */
 export function ModifiersProvider(props: ModifiersProviderProps): JSX.Element {
-  const context = useDayPicker();
-
-  const {
-    modifiers: customModifiers,
-    selected,
-    disabled,
-    hidden,
-    today,
-    fromDate,
-    toDate
-  } = {
-    ...context
-  };
+  const dayPicker = useDayPicker();
+  const selectMultiple = useSelectMultiple();
+  const selectRange = useSelectRange();
+
+  const internalModifiers: InternalModifiers = getInternalModifiers(
+    dayPicker,
+    selectMultiple,
+    selectRange
+  );
 
-  const multipleSelect = useSelectMultiple();
-  const rangeSelect = useSelectRange();
+  const customModifiers: CustomModifiers = getCustomModifiers(dayPicker);
 
   const modifiers: Modifiers = {
-    selected: toMatcherArray(selected),
-    disabled: toMatcherArray(disabled),
-    hidden: toMatcherArray(hidden),
-    today: [today],
-    range_end: [],
-    range_middle: [],
-    range_start: []
+    ...customModifiers,
+    ...internalModifiers
```

Comment:
I've moved out the logic from `ModifiersProvider` so I could test it. Now it should be clear that we are _not_ overriding the internal modifier, as the first version of the test was saying...